src/App/Fossa/LicenseScanner.hs

```diff
@@ -303,31 +308,89 @@ licenseScanSourceUnit ::
   ( Has Diagnostics sig m
   , Has (Lift IO) sig m
   , Has StickyLogger sig m
+  , Has Logger sig m
   , Has Exec sig m
   , Has ReadFS sig m
   , Has FossaApiClient sig m
   ) =>
+  Bool ->
   Path Abs Dir ->
   NonEmpty VendoredDependency ->
   m (NonEmpty Locator)
-licenseScanSourceUnit baseDir vendoredDeps = do
+licenseScanSourceUnit skippingSupported baseDir vendoredDeps = do
   uniqDeps <- dedupVendoredDeps vendoredDeps
-  -- At this point, we have a good list of deps, so go for it.
-  maybeArchives <- traverse (scanAndUploadVendoredDep baseDir) uniqDeps
-  archives <- fromMaybe NoSuccessfulScans $ NE.nonEmpty $ catMaybes $ NE.toList maybeArchives
-
-  -- archiveBuildUpload takes archives without Organization information. This orgID is appended when creating the build on the backend.
-  -- We don't care about the response here because if the build has already been queued, we get a 401 response.
-  finalizeLicenseScan $ ArchiveComponents $ NE.toList archives
 
   -- The organizationID is needed to prefix each locator name. The FOSSA API automatically prefixes the locator when queuing the build
   -- but not when reading from a source unit.
   orgId <- organizationId <$> getOrganization
 
+  -- make sure all revisions have their versions populated
+  uniqDepsWithVersions <- traverse (ensureVendoredDepVersion baseDir) uniqDeps
+  -- If skipping is supported, ask Core if any of these deps have already been scanned. If they have, skip scanning them.
+  (needScanningDeps, skippedDeps) <-
+    if skippingSupported
+      then findDepsThatNeedScanning uniqDepsWithVersions orgId
+      else pure (NE.toList uniqDepsWithVersions, [])
+
+  -- Debug logs giving info about which vendored deps were actually scanned
+  -- If none of the dependencies need scanning, then log that, but we still need to do `finalizeLicenseScan` so keep going
+  _ <- case (needScanningDeps, skippingSupported) of
+    (_, False) ->
+      logDebug "Skipping vendored dependency scans not supported on your FOSSA instance. All vendored dependencies will be scanned"
```

Comment:
[optional] I think referring to this as "skipping" is potentially confusing. Can we reword this in a more user friendly way?
Something like (but feel free to editorialize):
```haskell
logDebug "This version of the FOSSA service does not support enumerating previously scanned vendored dependencies."
logDebug "Performing a full scan of all vendored dependencies even if they have been scanned previously."
```

Some comments below are also rewordings, please also treat them as optional and/or modify as you see fit.